File: src/test/java/koorkah/TestOptional.java
package koorkah;

import org.junit.jupiter.api.Test;

class TestOptional extends BaseT
{

    @Test
    void test()
    {
        String expression = koorkah.toRegEx("`a` optional{`b`} `c`");
        assertRegexIs("ab?c", expression);
        assertRegexMatchesText(expression, "abc");
        assertRegexMatchesText(expression, "ac");
        assertRegexDoesntMatchText(expression, "abbc"); 
    }
    
    @Test
    void testComplex()
    {
        String expression = koorkah.toRegEx("`a` optional{ (`b` digit) } `c`");
        assertRegexIs("a(b\\d)?c", expression);
        assertRegexMatchesText(expression, "ab3c");
        assertRegexMatchesText(expression, "ac");
        assertRegexDoesntMatchText(expression, "abbc"); 
    }


}

File: src/python/test/TestOptional.py
import unittest
from Utils import Utils 
from Refi import Refi


class TestOptional(unittest.TestCase):

    def setUp(self):
        self.refi = Refi()
        self.utils = Utils()


    def test_(self):

        expression = self.refi.toRegEx("`a` optional{`b`} `c`")
        self.utils.assertRegexIs("ab?c", expression)
        self.utils.assertRegexMatchesText(expression, "abc")
        self.utils.assertRegexMatchesText(expression, "ac")
        self.utils.assertRegexDoesntMatchText(expression, "abbc") 
    
    def test_Complex(self):

        expression = self.refi.toRegEx("`a` optional{ (`b` digit) } `c`")
        self.utils.assertRegexIs("a(b\\d)?c", expression)
        self.utils.assertRegexMatchesText(expression, "ab3c")
        self.utils.assertRegexMatchesText(expression, "ac")
        self.utils.assertRegexDoesntMatchText(expression, "abbc") 


    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    


